from_mwt_ds/DataScience/vw_executor/tests/test_Vw.py

```diff
@@ -0,0 +1,426 @@
+        self.assertEqual(len(result.iloc[0]['!Job']), 2)
+        self.assertEqual(len(result.iloc[1]['!Job']), 2)
+
+        result = vw.train([self.input1, self.input2], pd.DataFrame(Grid([
+            '--cb_explore_adf --dsjson --epsilon 0.1',
+            {'#base': '--cb_explore_adf --dsjson', '--epsilon':  0.2}])))
+        self.assertEqual(isinstance(result, pd.DataFrame), True)
+        self.assertEqual(len(result), 2)
+        self.assertEqual(len(result.iloc[0]['!Job']), 2)
+        self.assertEqual(len(result.iloc[1]['!Job']), 2)
+
+        result = vw.test([self.input1, self.input2], pd.DataFrame(Grid([
+            '--cb_explore_adf --dsjson --epsilon 0.1',
+            {'#base': '--cb_explore_adf --dsjson', '--epsilon':  0.2}])))
+        self.assertEqual(isinstance(result, pd.DataFrame), True)
+        self.assertEqual(len(result), 2)
+        self.assertEqual(len(result.iloc[0]['!Job']), 2)
+        self.assertEqual(len(result.iloc[1]['!Job']), 2)
+
+        result = vw.train([self.input1, self.input2], pd.DataFrame(Grid([
+            {'#base': '--cb_explore_adf --dsjson', '--epsilon':  0.1},
+            {'#base': '--cb_explore_adf --dsjson', '--epsilon':  0.2}])))
+        self.assertEqual(isinstance(result, pd.DataFrame), True)
+        self.assertEqual(len(result), 2)
+        self.assertEqual(len(result.iloc[0]['!Job']), 2)
+        self.assertEqual(len(result.iloc[1]['!Job']), 2)
+
+        result = vw.test([self.input1, self.input2], pd.DataFrame(Grid([
+            {'#base': '--cb_explore_adf --dsjson', '--epsilon':  0.1},
+            {'#base': '--cb_explore_adf --dsjson', '--epsilon':  0.2}])))
+        self.assertEqual(isinstance(result, pd.DataFrame), True)
+        self.assertEqual(len(result), 2)
+        self.assertEqual(len(result.iloc[0]['!Job']), 2)
+        self.assertEqual(len(result.iloc[1]['!Job']), 2)
+
+        result = vw.train([self.input1, self.input2], pd.DataFrame(Grid({
+            '#base': ['--cb_explore_adf --dsjson'],
+            '--epsilon': [0.1, 0.2]
+        })))
+        self.assertEqual(isinstance(result, pd.DataFrame), True)
+        self.assertEqual(len(result), 2)
+        self.assertEqual(len(result.iloc[0]['!Job']), 2)
+        self.assertEqual(len(result.iloc[1]['!Job']), 2)
+
+        result = vw.test([self.input1, self.input2], pd.DataFrame(Grid({
+            '#base': ['--cb_explore_adf --dsjson'],
+            '--epsilon': [0.1, 0.2]
+        })))
+        self.assertEqual(isinstance(result, pd.DataFrame), True)
+        self.assertEqual(len(result), 2)
+        self.assertEqual(len(result.iloc[0]['!Job']), 2)
+        self.assertEqual(len(result.iloc[1]['!Job']), 2)
+
+    def test_e2e_test(self):
+        cache = Path('.vw_cache_test')
+        reset_cache_folder(cache)
+        
+        vw = Vw(cache, handlers = [])
+        stdout_cache = cache.joinpath('cacheNone')
+
+        result = vw.test(self.input1, '--cb_explore_adf --dsjson')
+        for l in result[0].stdout.raw:
```

Comment:
can we assert instead of print here?

Reply:
oh sure - going to remove it. that's me debugging ci failure - cannot reproduce it locally